src/components/InputTask.js

```diff
@@ -16,14 +16,23 @@ const InputTask = () => {
     event.preventDefault();
 
     try {
-      let uniqueId = id;
-      uniqueId++;
-      setId(uniqueId);
-      localStorage.setItem(uniqueId, description);
+      let uniqueKey = key;
```

Comment:
I am wondering if https://www.npmjs.com/package/uuid would be of assistance here?

Can you explain what the `uniqueKey` is for?

Reply:
uniqueKey is needed as localStorage saves as a key/value pair.
If the key isn't unique it will override the value previously assigned to that key.

I've put a check in to see if the key is already included in the localStorage array and if so choose another key based on the length of the localStorage array.

I realise this will cause problems once I start deleting tasks, as the localStorage array length will no longer be consecutive but current implementation is enough for my MVP, which just needs to create and. read tasks from storage.

I'll definitely need a solution like you suggested above once I flesh out the other CRUD operations.

Reply:
Thanks for the explain, only just saw this as I haven't looked at the mailbox my non Made Tech GitHub things go to for a while.